reference-service/src/main/java/com/transformuk/hee/tis/reference/service/api/FundingSubTypeResource.java

```diff
@@ -0,0 +1,199 @@
+    log.debug("REST request to update FundingSubType : {}", fundingSubTypeDto);
+    fundingSubTypeValidator.validate(fundingSubTypeDto);
+    if (fundingSubTypeDto.getUuid() == null) {
+      return createFundingSubType(fundingSubTypeDto);
+    }
+    FundingSubType fundingSubType = fundingSubTypeMapper.toEntity(fundingSubTypeDto);
+    fundingSubType = fundingSubTypeService.save(fundingSubType);
+    FundingSubTypeDto result = fundingSubTypeMapper.toDto(fundingSubType);
+    return ResponseEntity.ok()
+        .headers(
+            HeaderUtil.createEntityUpdateAlert(ENTITY_NAME, fundingSubTypeDto.getUuid().toString()))
+        .body(result);
+  }
+
+  /**
+   * DELETE  /funding-sub-type/:uuid : delete the "uuid" fundingSubType.
+   *
+   * @param uuid the uuid of the fundingSubTypeDto to delete
+   * @return the ResponseEntity with status 200 (OK)
+   */
+  @DeleteMapping("/funding-sub-type/{uuid}")
+  @PreAuthorize("hasAuthority('reference:delete:entities')")
+  public ResponseEntity<Void> deleteFundingSubType(@PathVariable UUID uuid) {
+    log.debug("REST request to delete FundingSubType : {}", uuid.toString());
+    fundingSubTypeService.deleteById(uuid);
+    return ResponseEntity.ok()
+        .headers(HeaderUtil.createEntityDeletionAlert(ENTITY_NAME, uuid.toString())).build();
+  }
+
+  /**
+   * GET  /funding-sub-type/:uuid : get the "uuid" fundingSubType.
+   *
+   * @param uuid the uuid of the fundingSubTypeDTO to retrieve
+   * @return the ResponseEntity with status 200 (OK) and with body the fundingSubTypeDTO, or with
+   *     status 404 (Not Found)
+   */
+  @GetMapping("/funding-sub-type/{uuid}")
+  public ResponseEntity<FundingSubTypeDto> getFundingSubType(@PathVariable UUID uuid) {
+    log.debug("REST request to get FundingType : {}", uuid.toString());
+    FundingSubType fundingSubType = fundingSubTypeService.findById(uuid)
+        .orElse(null);
+    FundingSubTypeDto fundingSubTypeDto = fundingSubTypeMapper.toDto(fundingSubType);
+    return ResponseUtil.wrapOrNotFound(Optional.ofNullable(fundingSubTypeDto));
+  }
+
+  /**
+   * GET  /funding-sub-types : get all funding sub types.
+   *
+   * @param pageable the pagination information. Please note: if you want to sort by id, use uuid
+   *                 instead
+   * @return the ResponseEntity with status 200 (OK) and the list of colleges in body
+   */
+  @ApiOperation(value = "Lists funding sub types",
+      notes = "Returns a list of funding sub types with support for "
+          + "pagination, sorting, smart search and column filters \n")
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message = "funding sub types list")})
+  @GetMapping("/funding-sub-types")
+  public ResponseEntity<List<FundingSubTypeDto>> getAllFundingSubTypes(
+      @ApiParam Pageable pageable,
+      @ApiParam(value = "any wildcard string to be searched")
+      @RequestParam(value = "searchQuery", required = false) String searchQuery,
```

Comment:
Can this be used to fetch "all sub types for a given parent type id"?

Reply:
No. I think that would be some changes on FundingTypeResource rather than FundingSubTypeResource.

Reply:
Cool, realised I think I'm jumping ahead to anther sub-task